### PR TITLE
Use theforeman/actions for GitHub Actions configuration

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

**Comment by ekohl** [comment]
@evgeni this is the only repo where I tried to also test with Foreman versions and archiving the lock files fails. I suppose the cache key isn't long enough and needs more :(

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,70 +1,34 @@
...
-        run: bundle exec rake 'plugin:assets:precompile[foreman_expire_hosts]'
-        env:
-          RAILS_ENV: production
+        foreman:
+          - 'develop'
+          - '3.9-stable'
+          - '3.8-stable'
+    uses: theforeman/actions/.github/workflows/foreman_plugin.yml@v0
+    with:
+      plugin: foreman_expire_hosts
```
Comment (evgeni) [inline .github/workflows/ci.yml:34]:
```suggestion
    with:
      plugin: foreman_expire_hosts
      foreman_version: ${{ matrix.foreman }}
```

**Comment by evgeni** [comment]
3.8 is failing, which I think is a test setup issue, not a code issue. should we just drop it from the matrix?

Edit: or pick https://github.com/theforeman/foreman/pull/9932 into 3.8?

**Comment by ekohl** [comment]
> Edit: or pick [theforeman/foreman#9932](https://github.com/theforeman/foreman/pull/9932) into 3.8?

I wonder why it even needs `PuppetRelatedHelper` in the tests. Perhaps we can just drop that.

**Comment by evgeni** [comment]
> > Edit: or pick [theforeman/foreman#9932](https://github.com/theforeman/foreman/pull/9932) into 3.8?
> 
> I wonder why it even needs `PuppetRelatedHelper` in the tests. Perhaps we can just drop that.

WFM

**Review by evgeni — APPROVED** [review]
ACK once this is green by either
- dropping 3.8 from the matrix
- picking that puppet change into 3.8
- dropping the puppethelper from the tests

:)

**Comment by evgeni** [comment]
now it fails on the branches that do have the helper :see_no_evil:

**Comment by ekohl** [comment]
How about conditionally including the helper? Let's see what CI says.

**Comment by ekohl** [comment]
That's now green.

**Review by evgeni — APPROVED** [review]
Whatever works

**Pull request merged**